subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/maven/MavenPomRelocationIntegrationTest.groovy

```diff
@@ -0,0 +1,146 @@
+<project>
+    <groupId>groupA</groupId>
+    <artifactId>projectA</artifactId>
+    <version>1.2</version>
+    <distributionManagement>
+        <relocation>
+            <groupId>groupB</groupId>
+            <artifactId>projectB</artifactId>
+        </relocation>
+    </distributionManagement>
+</project>
+"""
+
+        def newModule = mavenHttpRepo.module("groupB", "projectB", "1.2").publish()
+
+        and:
+        buildFile << """
+repositories { maven { url '${mavenHttpRepo.uri}' } }
+configurations { compile }
+dependencies { compile 'groupA:projectA:1.2' }
+task retrieve(type: Sync) {
+    into 'libs'
+    from configurations.compile
+}
+"""
+
+        and:
+        original.pom.expectGet()
+        newModule.pom.expectGet()
+        newModule.artifact.expectGet()
+
+        when:
+        run "retrieve"
+
+        then:
+        file("libs").assertHasDescendants("projectB-1.2.jar")
+    }
+
+    def "retrieve artifact when only groupId in <relocation>"() {
+        given:
+        def original = mavenHttpRepo.module("groupA", "projectA", "1.2").publishPom()
+        original.pomFile.text = """
+<project>
+    <groupId>groupA</groupId>
+    <artifactId>projectA</artifactId>
+    <version>1.2</version>
+    <distributionManagement>
+        <relocation>
+            <groupId>newProjectA</groupId>
+        </relocation>
+    </distributionManagement>
+</project>
+"""
+
+        def newModule = mavenHttpRepo.module("newProjectA", "projectA", "1.2").publish()
+
+        and:
+        buildFile << """
+repositories { maven { url '${mavenHttpRepo.uri}' } }
+configurations { compile }
+dependencies { compile 'groupA:projectA:1.2' }
+task retrieve(type: Sync) {
+    into 'libs'
+    from configurations.compile
+}
+"""
+
+        and:
+        original.pom.expectGet()
+        newModule.pom.expectGet()
+        newModule.artifact.expectGet()
+
+        when:
+        run "retrieve"
+
+        then:
+        file("libs").assertHasDescendants("projectA-1.2.jar")
+    }
+
+    def "fail when artifact in <relocation> doesn't exist"() {
+        given:
+        def original = mavenHttpRepo.module("groupA", "projectA", "1.2").publishPom()
+        original.pomFile.text = """
+<project>
+    <groupId>groupA</groupId>
+    <artifactId>projectA</artifactId>
+    <version>1.2</version>
+    <distributionManagement>
+        <relocation>
+            <groupId>notExist</groupId>
+            <artifactId>notExist</artifactId>
+        </relocation>
+    </distributionManagement>
+</project>
+"""
+
+        def newModule = mavenHttpRepo.module("notExist", "notExist", "1.2")
+
+        and:
+        buildFile << """
+repositories { maven { url '${mavenHttpRepo.uri}' } }
+configurations { compile }
+dependencies { compile 'groupA:projectA:1.2' }
+task retrieve(type: Sync) {
+    into 'libs'
+    from configurations.compile
+}
+"""
+
+        and:
+        original.pom.expectGet()
+        newModule.pom.expectGet()
+        newModule.artifact.expectHead()
```

Comment:
I think so...according to [here](https://github.com/gradle/gradle/blob/master/subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/resolver/ExternalResourceResolver.java#L201), an "orphan" artifact's existence is possible, so Gradle will try to `HEAD` the artifact if it couldn't find `pom`.